### PR TITLE
Adding some functionalities and a CMakeFiles.txt

### PR DESCRIPTION
**Pull request by rajdakin**
This commit would add some functionnalities, as adding a `CMakeLists.txt` or a red box around a voxel if you can't mine it (or yellow if you have no destructor tool equiped) and also fix some bugs that I have discovered.
This commit add also the capacity to BlackVoxel to launch when it is not installed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -190,9 +193,82 @@ time. In theory, it should work with minimal adaptations.
...
 Anyway, you need to have hardware 3D OpenGL support to run Blackvoxel.
 
+###### This will compile the last version:
+
+### Ubuntu
+
+1. Open a terminal type: "term" in ubuntu lancher or press 'ctrl-alt-t'
+2. Enter the following commands:
+```
+git clone https://github.com/rajdakin/Blackvoxel.git
```
Comment (asl97) [inline README.md]: Pointing to your own fork?
This pull request is obviously not ready to be merge back upstream.

**Comment by Blackvoxel** [comment]
Hello Rajdakin

First, we want to thank you for submitting such interesting work. As we understand by reading your code, you made a port on handheld device called Pandora.

But, we can't merge such massive pull request as doing so would make a mess in the project.

We are conscious that all this work take you a good amount of time and we are really sorry to not merge your pull request. This doesn't mean we won't use your work, though (see further). Here is some points explaining why we can't accept this pull request and what we can suggest you now :

- Some changes you made are interesting, but some other are less. This is a necessity in any project to make choices of what is wanted and what is not in order to maintain some coherence and direction. The best way is to discuss any change on our forum(http://forum.blackvoxel.com) before making it in order to avoid loosing time in work that are not accepted. That's also why any separate change should be the object of a separate pull request (we are a small team).

- We don't understand the purpose of some of the changes you made. Maybe some are needed for your handheld device. But we can't integrate some changes we don't know exactly the purpose. That's why integration of something needs discussion.

- Even a change we want to accept could need some rework and correction before being integrated. This rework could also be for us an occasion to improve our own code. A massive pull request is really difficult to rework and check.

- The developpement of Blackvoxel will continue for years, hopefully decades in a continuous developpement logic. But we wan't absolutely to avoid the "always broken" caveat. Blackvoxel is now a fairly stable game and we are working hard to preserve this stability despite adding a lot of new features. This mean we must check deeply any new code.

- Some point (like the use of a simple "make" rather than cmake or scons) was the result of a choice for some reasons. But we are not closed to discussion on this point and would be happy to hear your opinion on the question. We simply want to be sure to make the better choice for this project.

- Some of your changes show conflicts with 2.0 version. As an example, in 2.0 the whole game was completely rebalanced, so your robot patch should be suppressed.

Unfortunately, we weren't aware that you made this port. If we have known your project, we had helped you to make things in a way to make convergence easier.

Anyway we are very happy to discover your project. And the Pandora seem to be an very interesting device...

Rejecting this pull request don't mean we reject your work or won't make use of it. There is other ways to integrate these things with you. First, we encourage you to take contact with us on the forum(http://forum.blackvoxel.com). We would be happy if you could provide us some info on some points.

While reading your contributor list, you seem to be from France (or some french speaking country). As we are also french, if you prefer, we can speak in french.

Best regards
The Blackvoxel Team

**Comment by rajdakin** [comment]
I would like to discuss with you, but I can't send private messages... (I am logged in on the forum as rajdakin)
I'm closing this pull request since it's refuted.

**Comment by asl97** [comment]
Like what @Blackvoxel suggested, you should probably split this into separate pull request.

However, I was gonna suggest cherry picking some of the commit but it's a mess.
Not to mention, some of the commit are full of `EDIT` and `End of edit`.

It gonna be troublesome to split this.

**Pull request closed without merging**